File: solutions/javascript/002.js
function solution() {
    var nl;
    var n;
    var sum = 0;

    for (nl = 1, n = 2; n < 4e6; nl = n, n = new_n) {
        if (n % 2 === 0) {
            sum += n;
        }

        new_n = n + nl;
    }

    return sum;
}

if (require.main === module) {
    console.log(solution());
}

File: solutions/ruby/002.rb
def solution()
    nl = 1
    n = 2
    sum = 0

    while n < 4E6
        if n % 2 == 0
            sum += n
        end
        new_n = n + nl
        nl = n
        n = new_n
    end

    sum
end

if __FILE__ == $0
    puts solution()
end
